healthcheck/healthcheck.go

```diff
@@ -23,34 +25,41 @@ type HealthCheck struct {
 	Uptime                   time.Duration `json:"uptime"`
 	StartTime                time.Time     `json:"start_time"`
 	Checks                   []Check       `json:"checks"`
+	Started                  bool          `json:"-"`
+	Interval                 time.Duration `json:"-"`
 	Clients                  []*Client     `json:"-"`
 	CriticalErrorTimeout     time.Duration `json:"-"`
 	TimeOfFirstCriticalError time.Time     `json:"-"`
 	tickers                  []*ticker     `json:"-"`
 }
 
 // Create returns a new instantiated HealthCheck object. Caller to provide:
-// context and should utilise contextWithCancel
 // version information of the app,
 // criticalTimeout for how long to wait until an unhealthy dependent propagates its state to make this app unhealthy
 // interval in which to check health of dependencies
 // clients of type Client which contain a Checker function which is run to check the health of dependent apps
-func Create(ctx context.Context, version string, criticalTimeout, interval time.Duration, clients []*Client) HealthCheck {
-	tickers := newTickers(interval, clients)
+func Create(version string, criticalTimeout, interval time.Duration, clients []*Client) HealthCheck {
 
 	hc := HealthCheck{
+		Started:              false,
 		Clients:              clients,
 		Version:              version,
-		StartTime:            time.Now().UTC(),
 		CriticalErrorTimeout: criticalTimeout,
-		tickers:              tickers,
+		Interval:             interval,
 	}
-
-	hc.start(ctx)
-
 	return hc
 }
 
+// AddClient adds a provided client to the healthcheck
+func (hc *HealthCheck) AddClient(c *Client) {
+	log.Info("AddClient hit", log.Data{"client": c})
+	if hc.Started == false {
```

Comment:
I realize its the same number of lines, but it's more idiomatic to say 
```
if hc.Started {
 log.Error()
 return
}

hc.Clients = append(hc.Clients, c)
```

Because it keeps the happy path of this function in the primary line of sight